cspatients/tests.py

```diff
@@ -356,3 +446,212 @@ def test_right_patient_id_bad_dict(self):
+            patient_id
+        """
+        # Simulate posting the query string and getting a QueryDict
+        response_dict = QueryDict(query_string=self.post_query)
+
+        # Extract the dictionary using the get_rp_dict method
+        event_dict = get_rp_dict(response_dict)
+
+        # Test that it gets the appropriate keys and values
+
+        self.assertTrue(event_dict.__contains__("name"))
+        self.assertTrue(event_dict.__contains__("patient_id"))
+
+        # Test that it saves the right methods
+
+        self.assertTrue(event_dict["name"] == "Jane Doe")
+        self.assertTrue(event_dict["patient_id"] == "HLFSH")
+
+    def test_get_entrychanges_dict(self):
+        """
+            The Method when passed a context of entrychanges, must be able
+            to pick up the change category and the new value
+        """
+        # Simulate posting the query string and getting a QueryDict
+        response_dict = QueryDict(query_string=self.post_query)
+
+        # Extract the dictionary using the get_rp_dict method and
+        # entrychanges context
+        changes_dict = get_rp_dict(response_dict, context="entrychanges")
+
+        # Test that the change category and values are stored well
+        self.assertTrue(changes_dict.__contains__("name"))
+        self.assertTrue(changes_dict['name'] == "Nyasha")
+
+        # Test that the patient ID is kept well
+        self.assertTrue(changes_dict.__contains__("patient_id"))
+        self.assertTrue(changes_dict["patient_id"] == "HLFSH")
+
+
+# API Tests
+
+
+class RPEventTest(TestCase):
+
+    def setUp(self):
+        self.client = Client()
+        self.data = SAMPLE_RP_POST_DATA
+
+    def test_rpevent_saves_minimum_data_correctly(self):
+
+        delete_whole_table()
+        response = self.client.post(
+            "/cspatients/api/rpneepatiententry?secret=momkhulu",
+            self.data,
+        )
+        # Assert a correct response of 201
+        self.assertEqual(response.status_code, 201)
+
+        # Check that the Patient, PatientEntry been correctly saved
+        self.assertTrue(
+            Patient.objects.get(patient_id="HLFSH").name == "Jane Doe"
+            )
+        self.assertTrue(PatientEntry.objects.get(patient_id="HLFSH"))
+
+
+class RPPatientExistsTest(TestCase):
+
+    def setUp(self):
+        delete_whole_table()
+        self.client = Client()
+        self.good_data = SAMPLE_RP_POST_DATA
+        self.bad_data = SAMPLE_RP_POST_DATA_2
+
+        patient = Patient.objects.create(
+            name="Jane Doe",
+            patient_id="HLFSH",
+            age=20
+        )
+
+        PatientEntry.objects.create(
+            patient_id=patient
+        )
+
+    def test_finds_existing_patient(self):
+
+        response = self.client.post(
+            "/cspatients/api/rppatientexists",
+            self.good_data
+        )
+
+        self.assertEqual(response.status_code, 200)
+
+    def test_does_not_find_not_existing_patient(self):
```

Comment:
I think you might want to rename this to reflect the behavior that we expect. Something like `test_returns_404_for_non_existent_patient`